profile/profile.go

```diff
@@ -674,6 +674,39 @@ func numLabelsToString(numLabels map[string][]int64, numUnits map[string][]strin
 	return strings.Join(ls, " ")
 }
 
+// SetLabel sets the specified key to the specified value for all samples in the
+// profile.
+func (p *Profile) SetLabel(key string, value []string) {
+	for _, sample := range p.Sample {
+		if sample.Label == nil {
+			sample.Label = map[string][]string{}
+		}
+		sample.Label[key] = value
+	}
+}
+
+// RemoveLabel removes all labels associated with the specified key for all
+// samples in the profile.
+func (p *Profile) RemoveLabel(key string) {
+	for _, sample := range p.Sample {
+		if _, ok := sample.Label[key]; ok {
+			delete(sample.Label, key)
+		}
+	}
+}
+
+// HasLabel returns true if a sample has a label with indicated key and value.
+func (s *Sample) HasLabel(key, value string) bool {
+	if values, ok := s.Label[key]; ok {
```

Comment:
I think you could drop the "ok" check and just do
```
for _, v := range s.Label[key] {
```
as it will return nil slice if the key is missing which will immediately terminate the loop anyway.

Reply:
Done.